bart/sched/pelt.py

```diff
@@ -0,0 +1,464 @@
+
+            # Check if the task was running in the current PELT sample
+            running = task.isRunning(_us_to_ms(t_us))
+
+            # Assume the task was running for all the current PELT sample
+            active_us = Simulator._sample_us if running else 0
+            pelt_value = self._geomSum(pelt_value, active_us)
+
+            # Append PELT sample
+            sample = (_us_to_s(t_us), t_us/Simulator._sample_us, running, pelt_value)
+            samples.append(sample)
+
+            # Prepare for next sample computation
+            t_us += Simulator._sample_us
+
+        # Create DataFrame from computed samples
+        self._df = DataFrame(samples, columns=['Time', 'PELT_Interval', 'Running', 'pelt_value'])
+        self._df.set_index('Time', inplace=True)
+
+        # Keep track of the last task we computed the signal for
+        self.task = task
+
+        return self._df
+
+    def getStats(self, stats_start_s=0, stats_end_s=None):
+        """Compute the stats over a pre-computed PELT signal, considering only
+        the specified portion of the signal.
+
+        :param stats_start_s: the start of signal portion to build the stats for
+        :type  stats_start_s: float
+
+        :param stats_end_s: the end of signal portion to build the stats for
+        :type  stats_end_s: float
+
+        :return: :mod:`PELTStats' instance reporting the metrics of the PELT
+                 signal computed for the task specified in the last execution
+                 of getSignal.
+        """
+
+        # Validate input parameters
+        if self._df is None:
+            raise ValueError("no signal computed, run getSignal before")
+
+        # Intervals (in PELT samples) for the statistics to compute
+        stats_start_s = max(stats_start_s, self._df.index.min())
+        if stats_end_s:
+            stats_end_s = min(stats_end_s, self._df.index.max())
+        else:
+            stats_end_s = self._df.index.max()
+
+        df = self._df[stats_start_s:stats_end_s]
+
+        # Compute stats
+        _pelt_avg = self.task.pelt_avg
+        _pelt_init = self.init_value
+        _half_life = self.half_life_ms
+        _tmin = stats_start_s
+        _tmax = stats_end_s
+        _min = df.pelt_value.min()
+        _max = df.pelt_value.max()
+        _avg = df.pelt_value.mean()
+        _std = df.pelt_value.std()
+        _err = _avg - self.task.pelt_avg
+        _err_pct = None
+        if self.task.pelt_avg:
+            _err_pct = 100 * _err / self.task.pelt_avg
+
+        return PELTStats(
+            _us_to_s(self.start_us), _us_to_s(self.end_us),
+            _pelt_avg, _pelt_init, _half_life,
+            _tmin, _tmax, _min, _max,
+            _avg, _std, _err, _err_pct)
+
+
+################################################################################
+# Utility Functions
+################################################################################
+
+def _s_to_us(time_s, interval_us=1e3):
```

Comment:
@bjackman this should now be a better implementation of the time conversion functions which addresses this previous comments of yours:
https://github.com/ARM-software/bart/pull/65#discussion_r90434299

Have a look and let me know if the usage of the `interval_us` parameter is good enough instead of using a different variable name. The function is still named _s_to_us, but the default "units" can be tuned at call time to be 1024 instead of the default 1e3.